source/MaterialXCore/Document.cpp

```diff
@@ -182,47 +182,83 @@ void Document::initialize()
     setVersionIntegers(MATERIALX_MAJOR_VERSION, MATERIALX_MINOR_VERSION);
 }
 
-NodeDefPtr Document::addNodeDefFromGraph(const NodeGraphPtr nodeGraph, const string& nodeDefName, const string& node,
-                                         const string& version, bool isDefaultVersion, const string& group, const string& newGraphName)
+NodeDefPtr Document::addNodeDefFromGraph(const NodeGraphPtr nodeGraph, const string& nodeDefName,
+                                         const string& category, const string& newGraphName)
 {
+    if (category.empty())
+    {
+        throw Exception("Cannote create a nodedef without a category identifier");
+    }
+
     if (getNodeDef(nodeDefName))
     {
         throw Exception("Cannot create duplicate nodedef: " + nodeDefName);
     }
 
-    NodeGraphPtr graph = nodeGraph;
-    if (!newGraphName.empty())
+    if (getNodeGraph(newGraphName))
     {
-        if (getNodeGraph(newGraphName))
-        {
-            throw Exception("Cannot create duplicate nodegraph: " + newGraphName);
-        }
-        graph = addNodeGraph(newGraphName);
-        graph->copyContentFrom(nodeGraph);
+        throw Exception("Cannot create duplicate nodegraph: " + newGraphName);
     }
+    
+    // Create a new functional nodegraph, and copy over the
+    // contents from the compound nodegraph
+    NodeGraphPtr graph = addNodeGraph(newGraphName);
+    graph->copyContentFrom(nodeGraph);
+
+    for (auto graphChild : graph->getChildren())
+    {
+        graphChild->removeAttribute("xpos");
+        graphChild->removeAttribute("ypos");
+    }        
     graph->setNodeDefString(nodeDefName);
 
+    // Create a new nodedef and set it's category
     NodeDefPtr nodeDef = addChild<NodeDef>(nodeDefName);
```

Comment:
It might be clearer to call `Document::addNodeDef` here.  If you don't want to set the `type` of the NodeDef, you can set this argument to the empty string, only passing the `node` string to the constructor.

Reply:
I'll make the change.